File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/SlicedMushroom.java
package content.global.skill.production.cooking.recipe.topping.impl;

import content.global.skill.production.cooking.recipe.topping.ToppingRecipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.item.Item;

/**
 * The Sliced mushroom.
 */
public final class SlicedMushroom extends ToppingRecipe {

    private static final Item SLICED_MUSHROOMS = new Item(7080);

    private static final Item MUSHROOM = new Item(6004);

    private static final Item KNIFE = new Item(946);

    @Override
    public void mix(final Player player, final NodeUsageEvent event) {
        if (!player.getInventory().containsItem(KNIFE)) {
            player.getDialogueInterpreter().sendDialogue("You need a knife in order to slice up the mushrooms.");
            return;
        }
        super.mix(player, event);
    }

    @Override
    public int getLevel() {
        return 1;
    }

    @Override
    public double getExperience() {
        return 0;
    }

    @Override
    public Item getProduct() {
        return SLICED_MUSHROOMS;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{MUSHROOM};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{};
    }

    @Override
    public boolean isSingular() {
        return true;
    }
}

File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/SlicedMushroom.kt
package content.global.skill.production.cooking.recipe.topping.impl

import content.global.skill.production.cooking.recipe.topping.ToppingRecipe
import core.api.consts.Items
import core.api.inInventory
import core.api.sendDialogue
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.item.Item

class SlicedMushroom : ToppingRecipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (!inInventory(player, Items.KNIFE_946)) {
            sendDialogue(player, "You need a knife in order to slice up the mushrooms.")
            return
        }
        super.mix(player, event)
    }

    override fun getLevel(): Int {
        return 1
    }

    override fun getExperience(): Double {
        return 0.0
    }

    override fun getProduct(): Item {
        return SLICED_MUSHROOMS
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(MUSHROOM)
    }

    override fun getParts(): Array<Item> {
        return emptyArray()
    }

    override fun isSingular(): Boolean {
        return true
    }

    companion object {
        private val SLICED_MUSHROOMS = Item(Items.SLICED_MUSHROOMS_7080)
        private val MUSHROOM = Item(Items.MUSHROOM_6004)
    }
}

